### PR TITLE
Add and fix type hints in RecentConns

### PR DESCRIPTION
**Pull request by cschramm**
* Actually convert between stored and runtime items
* Add type hints
* Add a cast to resolve the circular type dependency issue
* Rename some internal methods and members

### CODE REVIEW

File: blueman/plugins/applet/RecentConns.py [hunk 1]
```diff
@@ -150,15 +145,13 @@ def on_manager_state_changed(self, state):
...
-                self.Adapters[str(adapter.get_object_path())] = str(adapter["Address"])
+            self._adapters = {str(adapter.get_object_path()): str(adapter["Address"])
+                              for adapter in adapters}
 
-            for i in reversed(self.items):
-                try:
-                    i["device"] = self.get_device_path(i)
-                except (AdapterNotFound, DeviceNotFound):
-                    pass
+            for i in self.items:
```
Comment (infirit) [inline blueman/plugins/applet/RecentConns.py:151]: You dropped ``reversed()`` here but left it in ``on_device_removed`` is that intentional? Also ``self.initialize()`` already sorts so I am confused why the reversed was there in the first place?

Comment (cschramm) [reply blueman/plugins/applet/RecentConns.py:151]: The reverse iteration is necessary there as the loop removes the current item from the list. It does not have any use here. It is a pre ac5eb6d8a3c09b116a07258d1a8914f667e197bc leftover where this loop actually used to remove as well.

**Comment by cschramm** [comment]
Quick notice: Removing the handler arguments in `on_item_activated` breaks some paths in `DBusService.connect_service`. I'm currently figuring out handler hell regarding service connections with type hints.

**Comment by infirit** [comment]
callback/handler hell 😒🥺

The removal of ``*args`` in reply is causing problems?

**Comment by cschramm** [comment]
Should break the handling for serial and network services.

**Comment by cschramm** [comment]
See #1329

**Pull request merged**